validation/markets-validator.js

```diff
@@ -503,7 +503,14 @@ function MetadataValidator(metadata, cityName) {
 
         request = request.request(url, function(response) {
             if (response.statusCode !== 200) {
-                asyncWarnings.push(new HttpResponseStatusIssue(cityName, response.statusCode, response.headers.location));
+                var location = response.headers.location;
```

Comment:
I would probably split this into multiple checks on `statusCode` instead of just comparing against `200`.

 We can expect the `300` range to have a `location` field but cannot do the same for the `400` range.

Kind of like request's [`raise_for_status`](http://docs.python-requests.org/en/master/_modules/requests/models/#Response.raise_for_status) works in python;